webapp/src/main/java/com/box/l10n/mojito/monitoring/QuartzPendingJobsReportingTask.java

```diff
@@ -0,0 +1,64 @@
+package com.box.l10n.mojito.monitoring;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+
+@Component
+@ConditionalOnProperty(value = "l10n.management.metrics.quartz-metrics.enabled", havingValue = "true")
+@ConditionalOnExpression("'${spring.datasource.url:}'.toLowerCase().contains('mysql')")
+public class QuartzPendingJobsReportingTask {
+
+    @Autowired
+    JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    MeterRegistry meterRegistry;
+
+    @Scheduled(fixedRateString = "${l10n.management.metrics.quartz-metrics.execution-rate}")
+    public void reportPendingJobs() {
+        fetchResults().forEach(this::reportResults);
+    }
+
+    private void reportResults(PendingJob pendingJob){
+        meterRegistry.gauge("quartz.pending.jobs",
+                            Tags.of("jobClass", pendingJob.jobClass, "jobGroup", pendingJob.jobGroup),
+                            pendingJob.count);
+    }
+
+    List<PendingJob> fetchResults(){
+        return jdbcTemplate.query(
+            "SELECT job_class_name, job_group, COUNT(*) FROM QRTZ_JOB_DETAILS d GROUP BY d.job_class_name, d.job_group",
+            (rs, num) -> new PendingJob(extractClassName(rs.getString(1)), rs.getString(2), rs.getLong(3))
+        );
+    }
+
+    private String extractClassName(String input) {
+        String[] parts = input.split("\\.");
+        return parts[parts.length-1];
```

Comment:
knit: indentation seems weird, do you apply intellij reformat?

Reply:
Yeah, its just intellij format

Reply:
strange, mine add space around `-`